akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/scaladsl/ClusterActorSet.scala

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright (C) 2009-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.cluster.sharding.typed.scaladsl
+
+import akka.actor.typed.ActorSystem
+import akka.actor.typed.Behavior
+import akka.actor.typed.Extension
+import akka.actor.typed.ExtensionId
+import akka.annotation.ApiMayChange
+import akka.annotation.DoNotInherit
+import akka.cluster.sharding.ShardRegion.EntityId
+import akka.cluster.sharding.typed.internal.ClusterActorSetImpl
+
+import scala.concurrent.duration.FiniteDuration
+import akka.util.JavaDurationConverters._
+import com.typesafe.config.Config
+
+object ClusterActorSetSettings {
+
+  def apply(system: ActorSystem[_]): ClusterActorSetSettings = {
+    apply(system.settings.config.getConfig("akka.cluster.actor-set"))
+  }
+
+  def apply(config: Config): ClusterActorSetSettings = {
+    val keepAliveInterval = config.getDuration("keep-alive-interval").asScala
+
+    new ClusterActorSetSettings(keepAliveInterval)
+  }
+
+}
+
+class ClusterActorSetSettings private[akka] (val keepAliveInterval: FiniteDuration) {
+
+  /**
+   * The interval each parent of the sharded set is pinged from each node in the cluster.
+   *
+   * Note: How the sharded set is kept alive may change in the future meaning this setting may go away.
+   */
+  @ApiMayChange
+  def withKeepAliveInterval(keepAliveInterval: FiniteDuration): ClusterActorSetSettings =
+    new ClusterActorSetSettings(keepAliveInterval)
+}
+
+object ClusterActorSet extends ExtensionId[ClusterActorSet] {
```

Comment:
@renatocaval didn't fully like the name `ClusterActorSet`, and I can agree that it's not fantastic. `ShardedWorkers` or `ShardedWorkersSet` were new suggestions.

Worker isn't too bad.

I see two key aspects:
* some fixed number of long running process that should be kept alive
* it doesn't receive message as input but has some other source of input for the work

The latter is a restriction that I think is a bit unfortunate, because I can see that it would be useful to also send messages to them.

Thinking of "daemon process" in unix. `ShardedDaemonProcess`, `SlicedDaemonProcess`, `SlicedDaemonWorker` ?

Reply:
Other previous naming ideas (from internal discussion):

* (sharded) keep-alive
* multipleton (joke but not completely off)
* sharded singletons
* singleton set
* cluster set
* sharding topology

Reply:
Isn't "worker" a bit much up to the use case, it fits for projections but not other uses?

They could also receive messages as input, just that they'd have to pull or register themselves somewhere.

Reply:
I also think that that 'worker' is a little bit overload. So maybe not the best name. 

I like `ShardedDaemonProcess` because if expresses the idea that is something that keeps running on the background and is distributed.

The `sliced` prefix I find less interesting. Shard is the term we use already and slice makes me think on collection API.

Reply:
Let's go with `ShardedDaemonProcess` then.

Reply:
good, I like it